crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -1071,7 +1085,16 @@ where
         // internally-set timestamp in the inherents actually matches the slot set in the seal
         // and root blocks in the inherents are set correctly.
         if let Some(extrinsics) = extrinsics {
-            // TODO: Verify the fraud proof is correct if the extrinsic `submit_fraud_proof` is found.
+            // Ensure the fraud proof is valid if the `submit_fraud_proof` extrinsic is included.
```

Comment:
I agree this'll work, but it seems to be completely orthogonal to consensus. What do you think about moving it to a different `BlockImport` in a separate crate, so we can keep these concerns separate?

Reply:
Broadly speaking, I think it's related to the consensus as you're dependent on the global consensus between the nodes in the network, but I agree with moving it to `subspace-fraud-proof` which provides a `BlockImport` checking the fraud proof, which sounds clearer. Will update.

Reply:
I was thinking of a separate `sc-` crate potentially

Reply:
I don't have a strong opinion, then perhaps `sc-consensus-fraud-proof`?

Reply:
Looks fine to me